app/cloud_tasks/cloud_task_publishers.py

```diff
@@ -0,0 +1,55 @@
+from google import auth
+from google.cloud.tasks_v2 import CloudTasksClient
+from google.cloud.tasks_v2.types.task import Task
+from structlog import get_logger
+
+from app.settings import SUBMISSION_CONFIRMATION_CLOUD_FUNCTION_NAME
+
+from .exceptions import CloudTaskCreationFailed
+
+logger = get_logger(__name__)
+
+
+class CloudTaskPublisher:
+    def __init__(self):
+        self._client = CloudTasksClient()
+
+        _, self._project_id = auth.default()
+
+    def _create(self, task: dict, queue_name: str) -> Task:
+        logger.info("creating cloud task")
+
+        self._parent = self._client.queue_path(
+            self._project_id, "europe-west2", queue_name
+        )
+
+        service_account_email = (
+            f"cloud-functions@{self._project_id}.iam.gserviceaccount.com"
+        )
+
+        url = f"https://europe-west2-{self._project_id}.cloudfunctions.net/{SUBMISSION_CONFIRMATION_CLOUD_FUNCTION_NAME}"
+
+        keys = {
+            "url": url,
+            "oidc_token": {"service_account_email": service_account_email},
+        }
+
+        task.update(keys)
```

Comment:
Currently, these are at the wrong level and fails with `Protocol message Task has no "url" field.` Which also indicate we are missing a test for the task payload structure.
```py
{
'http_request': {}
'url': '', 
'oidc_token': {}
}
```
`url` and `odic_token` should be nested under `http_request`.

Worth review my other comment of where the task definition should live first.